src/libraries/System.Net.Http/tests/FunctionalTests/HttpContentTest.cs

```diff
@@ -509,6 +509,98 @@ public async Task LoadIntoBufferAsync_ThrowIOExceptionInOverriddenAsyncMethod_Th
             Assert.IsType<IOException>(ex.InnerException);
         }
 
+        [Fact]
+        public async Task LoadIntoBufferAsync_Buffered_IgnoresCancellationToken()
+        {
+            string content = Guid.NewGuid().ToString();
+
+            await LoopbackServer.CreateClientAndServerAsync(
+                async uri =>
+                {
+                    using HttpClient httpClient = CreateHttpClient();
+
+                    HttpResponseMessage response = await httpClient.GetAsync(
+                        uri,
+                        HttpCompletionOption.ResponseContentRead);
+
+                    var cts = new CancellationTokenSource();
+                    cts.Cancel();
+
+                    await response.Content.LoadIntoBufferAsync(cts.Token);
+                },
+                async server =>
+                {
+                    await server.AcceptConnectionSendResponseAndCloseAsync(content: content);
+                });
+        }
+
+        [Fact]
+        public async Task LoadIntoBufferAsync_Unbuffered_CanBeCanceled_AlreadyCanceledCts()
+        {
+            await LoopbackServer.CreateClientAndServerAsync(
+                async uri =>
+                {
+                    using HttpClient httpClient = CreateHttpClient();
+
+                    HttpResponseMessage response = await httpClient.GetAsync(
+                        uri,
+                        HttpCompletionOption.ResponseHeadersRead);
+
+                    var cts = new CancellationTokenSource();
+                    cts.Cancel();
```

Comment:
If you're not going to check the identity of the token, this can just be `new CancellationToken(true)`. But if we expect the resulting exception to contain this token, then we should also be validating that after the below assert.

Reply:
Tests updated accordingly.